test/e2e/framework/framework.go

```diff
@@ -75,9 +74,8 @@ type Framework struct {
 	ClientSet                        clientset.Interface
 	KubemarkExternalClusterClientSet clientset.Interface
 
-	InternalClientset *internalclientset.Clientset
-	AggregatorClient  *aggregatorclient.Clientset
-	DynamicClient     dynamic.Interface
+	AggregatorClient *aggregatorclient.Clientset
```

Comment:
the aggregated apiserver imports, `clientset_generated/clientset` so we still have the internal client set, dependency wise.

but then it's not used too:
```
~/go/src/k8s.io/kubernetes/test$ grep AggregatorClient * -rnI
e2e/apimachinery/aggregator.go:72:		aggrclient = f.AggregatorClient
e2e/apimachinery/aggregator.go:106:	aggrclient := f.AggregatorClient
e2e/framework/framework.go:79:	AggregatorClient  *aggregatorclient.Clientset
e2e/framework/framework.go:185:		f.AggregatorClient, err = aggregatorclient.NewForConfig(config)
```

i think it can be removed too.

DynamicClient on the other hand seems fine because that's "k8s.io/client-go/dynamic".

Reply:
That seems a nice idea.
According to https://cs.k8s.io/?q=AggregatorClient&i=nope&files=&repos= that still is used in some repos but the fact is e04b91facf180c17557a44e8e462858ea2936301 is not applied to these repos yet.
So it is nice to remove AggregatorClient also.
I don't have a strong opinion we need to remove it on this PR or another PR.

Reply:
The aggregator clientset is imported from `k8s.io/kube-aggregator` though, would we still consider that an internal client set?

I prefer to do aggregator removal in a separate PR, someone else can take that one if they want :)

Reply:
@oomichi 
> So it is nice to remove AggregatorClient also.

+1

@andrewsykim 

> Sorry if I'm missing something here. The aggregator clientset is imported from k8s.io/kube-aggregator (staged directory) though, would we still consider that an internal client set?

from my understanding the aggregated apiserver *uses* the internal clientset.
e.g. `aggregatorclient "k8s.io/kube-aggregator/pkg/client/clientset_generated/clientset"`

waiting on @timothysc to comment on this.

> I prefer to do aggregator removal in a separate PR, someone else can take that one if they want :)

ok, sounds good.

/lgtm
/hold

Reply:
+1 aim to remove all deps on any transitive deps that yank in the internal client.